vspace/src/main/java/edu/asu/diging/vspace/core/file/impl/StorageEngine.java

```diff
@@ -21,38 +26,56 @@
 @PropertySource({"classpath:config.properties", "${appConfigFile:classpath:}/app.properties"})
 public class StorageEngine implements IStorageEngine {
 
-	@Value("${uploads_path}")
-	private String path;
-	
-	/* (non-Javadoc)
-	 * @see edu.asu.diging.vspace.core.file.impl.IStorageEngine#storeFile(byte[], java.lang.String, java.lang.String)
-	 */
-	@Override
-	public String storeFile(byte[] fileContent, String filename, String directory) throws FileStorageException {
-		File parent = new File(path + File.separator + directory);
-		if (!parent.exists()) {
-			parent.mkdir();
-		}
-		File file = new File(parent.getAbsolutePath() + File.separator + filename);
+    @Value("${uploads_path}")
+    private String path;
+
+
+    /* (non-Javadoc)
+    * @see edu.asu.diging.vspace.core.file.impl.IStorageEngine#storeFile(byte[], java.lang.String, java.lang.String)
+    */
+    @Override
+    public String storeFile(byte[] fileContent, String filename, String directory) throws FileStorageException {
+        
+        
+        File file = getFile(directory, filename);
         BufferedOutputStream stream;
         try {
             stream = new BufferedOutputStream(new FileOutputStream(file));
         } catch (FileNotFoundException e) {
-            throw new FileStorageException("Could not store file.", e);
+	    throw new FileStorageException("Could not store file.", e);
         }
         try {
             stream.write(fileContent);
             stream.close();
         } catch (IOException e) {
             throw new FileStorageException("Could not store file.", e);
         }
-        
+
         return directory;
-	}
-	
-	@Override
-	public byte[] getMediaContent(String directory, String filename) throws IOException {
-		File fileObject = new File(path + File.separator + directory + File.separator + filename);
+    }
+
+    @Override
+    public byte[] getImageContent(String directory, String filename) throws IOException {
```

Comment:
if this method just calls another method with the exact same parameters, you don't need two.